cmd/var.go

```diff
@@ -0,0 +1,126 @@
+package cmd
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/driusan/dgit/git"
+)
+
+func getLogVar(c *git.Client, config *git.GitConfig, key string, envVar string) string {
+	if envVar != "" {
+		return envVar
+	}
+
+	if key == "GIT_AUTHOR_IDENT" {
+		name := os.Getenv("GIT_AUTHOR_NAME")
+		email := os.Getenv("GIT_AUTHOR_EMAIL")
+		date := os.Getenv("GIT_AUTHOR_DATE")
+		if name != "" && email != "" && date != "" {
+			return fmt.Sprintf("%s <%s> %s", name, email, date)
+		}
+	}
+
+	if key == "GIT_COMMITTER_IDENT" {
+		name := os.Getenv("GIT_COMMITTER_NAME")
+		email := os.Getenv("GIT_COMMITTER_EMAIL")
+		date := os.Getenv("GIT_COMMITTER_DATE")
+		if name != "" && email != "" && date != "" {
+			return fmt.Sprintf("%s <%s> %s", name, email, date)
+		}
+	}
+
+	if key == "GIT_AUTHOR_IDENT" || key == "GIT_COMMITTER_IDENT" {
+		t := time.Now()
+		person := c.GetAuthor(&t)
+		return fmt.Sprintf("%s <%s> %v %s", person.Name, person.Email, time.Now().Unix(), time.Now().Format("-0700"))
+	} else if key == "GIT_EDITOR" {
+		coreEditor, _ := config.GetConfig("core.editor")
+		if coreEditor != "" {
+			return coreEditor
+		}
+
+		visual := os.Getenv("VISUAL")
+		if visual != "" {
+			return visual
+		}
+
+		editor := os.Getenv("EDITOR")
+		if editor != "" {
+			return editor
+		}
+
+		return "vi"
```

Comment:
In package git we default to ed, not vi

Reply:
Ah, ok. I can change that. I was going on this documentation that seemed to suggest that vi was the usual default. https://git-scm.com/docs/git-var

Ed because it's available on Plan 9?

Reply:
Partly because it's on Plan 9, partly because ed is the standard unix text editor,  and partly as a joke.